core/src/avm2/object/qname_object.rs

```diff
@@ -43,15 +48,15 @@ pub struct QNameObjectData<'gc> {
     /// All normal script data.
     base: ScriptObjectData<'gc>,
 
-    /// The QName name this object is associated with.
-    qname: Option<QName<'gc>>,
+    /// The Multiname this object is associated with.
+    name: Multiname<'gc>,
 }
 
 impl<'gc> QNameObject<'gc> {
-    /// Box a QName into an object.
-    pub fn from_qname(
+    /// Box a Multiname into an object.
+    pub fn from_name(
         activation: &mut Activation<'_, 'gc>,
-        qname: QName<'gc>,
+        name: impl Into<Multiname<'gc>>,
```

Comment:
IMO let's keep being explicit here - the only other thing that can be turned into a Multiname are QNames, and I'd prefer them to be obvious.